Adding Chef Infra Server to Chef Automate (#4826)
chef/automate: components/docs-chef-io/content/automate/chef_infra_server.md

Add the Name, Projects, Admin User, and Admin Key. Always use the exact details that were provided at the time of setting up _Knife_. Copy the contents of the `~/.chef/USER.pem` file and paste it into the _Admin Key_ field. Once done select **Add Chef Organization** to add the Chef Infra Server.

Select any Chef Organization to view the _Objects_.

## Access Chef Infra Server Objects

The Chef Infra Server Objects can be managed from the Chef Automate. In Automate Infra views, following objects can be managed:

- Cookbooks
- Roles
- Environments
- Data Bags
- Clients

Refer to the image below:

{{< figure src="/images/automate/chef-infra-server-objects.png" width="500" alt="Chef Infra Server Objects">}}

### Cookbooks

A [cookbook]({{< relref "cookbooks" >}}) is the fundamental unit of configuration and policy distribution. Include recipes and other files, components, or directories.

### Roles

[Roles]({{< relref "roles" >}}) let you define patterns and processes that exist across nodes in an organization as belonging to a single job function. Each role consists of zero (or more) attributes and a run-list.

### Environments

> An [environment]({{< relref "environment" >}}) can be used to map an organization's real-life workflow to what can be configured and managed when using Chef Infra. This mapping is accomplished by setting attributes and pinning cookbooks at the environment level.

Reviewer: This should make it build correctly.

```suggestion
An [environment]({{< relref "environments" >}}) can be used to map an organization's real-life workflow to what can be configured and managed when using Chef Infra. This mapping is accomplished by setting attributes and pinning cookbooks at the environment level.
```

+++
title = "Chef Infra Server"

weight = 20
draft = false

gh_repo = "automate"
[menu]
  [menu.automate]
    title = "Chef Infra Server"
    parent = "automate/infrastructure"
    identifier = "automate/infrastructure/chef_infra_server.md Chef Infra Server"
    weight = 20
+++

## Overview

The _Chef Infra Server_ page shows all the Chef Infra Servers connected to Chef Automate. The Chef Infra Server user interface can be used to view and manage all the Chef Infra Server Objects like:

- Cookbooks
- Roles
- Environments
- Data Bags
- Clients

> The _Chef Infra Server_ acts as a hub for configuration data. The Chef Infra Server stores _cookbooks_, the policies that are applied to _nodes_, and metadata that describes each registered node that is being managed by Chef Infra Client. Nodes use Chef Infra Client to ask the Chef Infra Server for configuration details, such as recipes, templates, and file distributions. Chef Infra Client then does as much of the configuration work as possible on the nodes themselves (and not on the Chef Infra Server). This scalable approach distributes the configuration effort throughout the _organization_.

Reviewer 2: This is a good addition! Can you simplify it a bit more?

To add the previously created organization to the Infrastructure, click **Add Chef Organization** button. Clicking on the button opens a popup menu as shown below:

![Add Chef Organization Popup Menu](/images/automate/add-chef-organization-popup-menu.png)

Add the Name, Projects, Admin User, and Admin Key. Always use the exact details that were provided at the time of setting up _Knife_. Copy the contents of the `~/.chef/USER.pem` file and paste it into the _Admin Key_ field. Once done click **Add Chef Organization**. The desired organization will be added to the Chef Server.

Click on any Chef Organization to view the _Objects_.

## Access Chef Infra Server Objects

The Chef Infra Server Objects can be managed from the Chef Automate. In Automate Infra views, following objects can be managed:

- Cookbooks
- Roles
- Environments
- Data Bags
- Clients

Refer to the image below:

![Chef Infra Server Objects](/images/automate/chef-infra-server-objects.png)

### Cookbooks

> A [cookbook](https://docs.chef.io/cookbooks/) is the fundamental unit of configuration and policy distribution. Include recipes and other files, components, or directories.

Reviewer: The links should use relative URLs. Ideally with [relref](https://gohugo.io/content-management/cross-references/) so we'll know if a link breaks if page changes or gets deleted. So `[cookbook](https://docs.chef.io/cookbooks/)` should be formatted as `[cookbook]({{< relref "cookbooks" >}})`.

### Roles

[Roles](https://docs.chef.io/roles/) let you define patterns and processes that exist across nodes in an organization as belonging to a single job function. Each role consists of zero (or more) attributes and a run-list.

### Environments

> An [environment](https://docs.chef.io/environments/) can be used to map an organization’s real-life workflow to what can be configured and managed when using Chef Infra. This mapping is accomplished by setting attributes and pinning cookbooks at the environment level.

Reviewer 2: Change `organization’s`  to `organization's"

I think you may have smart quotes enabled, if so, change them to straight quotes. This will save you a lot of frustration. https://support.microsoft.com/en-us/office/smart-quotes-in-word-702fc92e-b723-4e3d-b2cc-71dedaf2f343

+++

> title = "Client Runs"

Reviewer 2: Chef Infra Server

date = 2021-03-12T12:01:58-07:00
draft = false

gh_repo = "automate"
[menu]
  [menu.automate]
    title = "Chef Infra Server"
    parent = "automate"
    identifier = "automate/chef_infra_server.md Chef Infra Server"
    weight = 60
+++

## Overview

> The __Chef Server__ page shows all the Chef Infra Servers connected to Chef Automate. The __Chef Server__ user interface can be used to view and manage all the Chef Infra Server Objects like:

Reviewer 2: We need a thorough overview of the Chef Infra Server page and its contents before moving onto the actions. See how it was handled on other docs pages.

- Cookbooks
- Roles
- Environments
- Data Bags
- Clients

## Connect to the Chef Infra Server

To start making use out of the Chef Infra Server, firstly deploy and run the [Chef Automate](https://docs.chef.io/automate/install/#installation-guide).

**Note:** No explicit configuration is required to install the Automate Infra views.

Login to Chef Automate using the credentials. Once logged in, the page looks like as shown below:

> ![Chef Automate](/images/automate/chef-automate-on-chef-infra-page.png)

Reviewer 2: The picture is of the event feed, not Infra Server content

> ![Chef Automate](/images/automate/chef-automate-on-chef-infra-page.png)

Author: This won't be possible from my end I think.

Click on **Infrastructure** located at the top bar of the page. Refer to the image below:

> ![Chef Automate with Highlighted Infrastructure](/images/automate/chef-automate-infrastructure.png)

Reviewer 2: Also another image of the Event feed?

Ok, I think I get what you're trying to do. It is OK to say "To view your Chef Infra Servers in Automate, select **Infrastructure > Chef Infra Server**" or something similar. We can assume that the have an Automate instance running and know how to navigate tabs in a UI.

Click on the **Chef Servers** located at the left panel of the page. The page looks like as shown below:

![Chef Infra Server](/images/automate/chef-server-page.png)

The above image is the **Chef Server** page which contains all the Chef Infra Servers connected to the Chef Automate. The Chef Infra Servers listed are the ones that are already configured and added to the Chef Automate. To create a new Infra Server click [here](https://docs.chef.io/automate/infra_server/).

To add the created Chef Server to the Infrastructure, click on **Add Chef Server** button. Refer to the image below:

> ![Add Chef Server](/images/automate/add-chef-server-button.png)

Reviewer: I noticed that this image and a few others are quite large on the page, especially for what they're showing. You might try the [`figure` shortcode](https://gohugo.io/content-management/shortcodes/#figure) which has a width parameter.

```suggestion
{{< figure src="/images/automate/add-chef-server-button.png" width="500" alt="Image showing button for adding new Chef Infra Server.">}}
```

> Clicking on the **Add Chef Server** button opens a popup menu as shown below:

Reviewer 2: Select **Add Chef Server** and ...
Using "select" applies to all instances of "click on". 

Explain the fields that the user needs to complete. Emphasize the actions that a user needs to take instead of the UI. If you feel we need to mention the popup, use "Selecting **Add Chef Sever** opens a form" or "helper window", which we tried using in the past (I'm not a fan of that one, but it works).

Use example company from the docs style guide (4th Cafe, 4thcafe.com, ip: 192.0.2.0) (The ip for example domains in [rfc5737](https://tools.ietf.org/html/rfc5737)).

![Add Chef Server Popup Menu](/images/automate/add-chef-server-popup-menu.png)

Mention the correct Name, FQDN, and IP Address of the system. Click on **Add Chef Server**. The desired server gets added to the Chef Infra Server.

> **Note:** If the mentioned FQDN and IP Address is incorrect, the box displays an error as shown below:

Reviewer 2: Make this note more direct, "Adding invalid FQDNs or IP address cause an error:"

Or something similar

![Add Chef Server Popup Menu](/images/automate/add-chef-server-popup-menu-with-error.png)

Click on the listed Servers to view the details of the specific **Chef Server**.

## Connect to the Chef Infra Server Organization

Selecting a specific added Chef Server opens the list of organizations in the Server. Refer to the image below:

![Chef Infra Server Organization](/images/automate/chef-server-organization.png)

> The Organization in the Chef server listed are the ones which are already created in that specific server. To create a new Organization in the Server click [here](https://docs.chef.io/automate/infra_server/#set-up-the-chef-infra-server).

Reviewer 2: Revise "The Organization in the Chef server listed are the ones which are already created in that specific server" to be shorter & more direct.

Reviewer 2: "To create a new Organization in the Server click [here]" makes it sound like I could add a server in the UI. Since that's not the case. let's be clear that they need to do it on the server from the command line.

> To add the previously created organization to the Infrastructure, click on the **Add Chef Organization** button. Refer to the image below:

Reviewer 2: ```suggestion
To add an organization that already exists on your Chef Infra Server to the Infrastructure tab, select **Add Chef Organization** button.
```
And then do what?

![Add Chef Organization](/images/automate/add-chef-organization-button.png)

Clicking on the **Add Chef Organization** button opens a popup menu as shown below:

![Add Chef Organization Popup Menu](/images/automate/add-chef-organization-popup-menu.png)

> Add the Name, Projects, Admin User, and Admin Key. Always use the exact details that were provided at the time of setting up **Knife**. Copy the contents of the `~/.chef/USER.pem` file and past it into the **Admin Key** field. Once done click **Add Chef Organization**. The desired organization will be added to the Chef Server.

Reviewer 2: We need to clairify this a bit, but I don't have the right words at the moment. I will update with a suggestion tomorrow.

Click on any Chef Organization to view the **Objects**.

## Access Chef Infra Server Objects

> The **Chef Infra Server Objects** can be managed from the Chef Automate. In Automate Infra views, following objects can be managed:

Reviewer 2: ```suggestion
The **Chef Infra Server Objects** can be managed from the Chef Automate. You can use **Infrastructure** to manage:
```

- Cookbooks
- Roles
- Environments
- Data Bags
- Clients

Refer to the image below:

![Chef Infra Server Objects](/images/automate/chef-infra-server-objects.png)

### Cookbooks

> A cookbook is the fundamental unit of configuration and policy distribution. Cookbooks also comprise recipes and other optional components or directories. Click [here](https://docs.chef.io/cookbooks/) to know more about Cookbooks.

Reviewer 2: ```suggestion
A cookbook is the fundamental unit of configuration and policy distribution. Cookbooks include recipes and other files, components, or directories. Click [here](https://docs.chef.io/cookbooks/) to know more about Cookbooks.
```

### Roles

> A role is a way to define certain patterns and processes that exist across nodes in an organization as belonging to a single job function. Each role consists of zero (or more) attributes and a run-list. Click [here](https://docs.chef.io/roles/) to know more about Roles.

Reviewer 2: Consider "Roles let you define patterns and...." 
Use named links instead of `[here]`

### Environments

> An environment is a way to map an organization’s real-life workflow to what can be configured and managed when using Chef Infra. This mapping is accomplished by setting attributes and pinning cookbooks at the environment level. Click [here](https://docs.chef.io/environments/) to know more about Environments.

Reviewer 2: I think we should avoid "is a way" if we can. 
Use named links instead of `[here]`

### Data Bags

Data bags store global variables as JSON data. Data bags are indexed for searching and can be loaded by a cookbook or accessed during a search. Click [here](https://docs.chef.io/data_bags/) to know more about Data Bags.

### Clients

Chef Infra Clients provide secure API access to the Chef Infra Server.

## Troubleshoot

> While fetching any object like cookbooks or any, if it raises the `Could not get cookbooks: organization 'no-org' does not exist` error, the provided organization does not exist on Chef Infra Server. The error looks like as shown below:

Reviewer 2: Revise this sentence ("or any...").

![Could not get cookbooks: Organization](/images/automate/could-not-get-cookbooks-organization.png)

> Create the Chef Organization either from `knife opc org create` or using `chef-server-ctl org-create`. Once done add the exact details to fetch the objects.

Reviewer 2: ```suggestion
Create the Chef Organization using the knife command, `knife opc org create` or the Chef Infra Server CLI,  `chef-server-ctl org-create`. Then add the exact details to fetch the objects.
```
"exact details" is unclear to me
